src/query/traits.rs

```diff
@@ -88,3 +91,95 @@ where
         QuerySelect::column_as(self, col, alias)
     }
 }
+
+/// Insert query Trait
+pub trait InsertTrait<A>: Sized
```

Comment:
why do we need a new Trait here? it is not strictly necessary?

Reply:
Not a must, we just want to share the common methods between `Insert` and `TryInsert`. Methods such as `one`, `many`, `add`...etc.

Reply:
They can all implement the same methods. I don't see `InsertTrait` being used in a type signature anywhere so I think it is not needed.

Reply:
My concern is code duplication. Implement the same method twice, for two struct, `Insert` and `TryInsert`.

We can use macro, but I think `InsertTrait` to define shared methods might be better. Both are okay for me.

Reply:
Yes. technically `TryInsert` contains an `Insert` so any implementation can simply be delegated. I am not seeing where's the potential saving in code duplication.